### ISSUE TITLE
conda fails to update pythonpath with pymol location

### ISSUE THREAD

**Issue by bionicles** [schrodinger/pymol-open-source#31]
conda install -c schrodinger pymol adds the GUI command for running from bash, but we need to script multiple invocations of pymol cmd 

the conda installer doesnt properly update the pythonpath ... when you run python 3.7 after conda install, and try

from pymol import cmd
it reads, "ImportError: No module named pymol"

great, so i added /usr/local/Cellar/pymol/2.2.0/libexec/lib/python3.7/site-packages/pymol to PYTHONPATH in .bash_profile

now it still fails, because pymol/cmd.py can't find pymol ? how does that make sense?

well, there's a pymol_path folder inside site-packages/pymol. why is pymol_path not actually the path to pymol? plus, even worse, there's site-packages/pymol2/cmd2.py ??? why are there two cmd files and which one are we supposed to use in python scripts?

export PYTHONPATH="/usr/local/Cellar/pymol/2.2.0/bin"
doesn't work

export PYTHONPATH="/usr/local/Cellar/pymol/2.2.0/libexec/lib/python3.7/site-packages/pymol:$PYTHONPATH"
doesn't work

export PYTHONPATH="/usr/local/Cellar/pymol/2.2.0/libexec/lib/python3.7/site-packages/pymol/pymol_path:$PYTHONPATH"
still doesn't work

conda install should tell python where pymol lives so every user doesn't have to waste hours figuring hour how to properly tell python where to find the thing they just installed

https://pymolwiki.org/index.php/PYMOL_PATH not helpful

what the hell do i have to do to get PYTHON to find PYMOL ?

cat /usr/local/bin/pymol
```
#!/bin/sh
[ "$DISPLAY" == "" ] && export DISPLAY=":0.0"
export PYMOL_PATH="/usr/local/Cellar/pymol/2.2.0/libexec/lib/python3.7/site-packages/pymol/pymol_path"
"/usr/local/opt/python/bin/python3.7" "/usr/local/Cellar/pymol/2.2.0/libexec/lib/python3.7/site-packages/pymol/__init__.py" "$@"
```
?

**Comment by bionicles** [comment]
```
cd /usr/local/opt/pymol/libexec/lib/python3.7/site-packages/pymol/pymol_path/examples/devel
python importing.py
```
> AttributeError: module 'string' has no attribute 'split'

**Comment by bionicles** [comment]
```
(neuromax-2019) /usr/local/opt/pymol/libexec/lib/python3.7/site-packages/pymol/pymol_path/examples/launching $ python launch_no_tk.py
```

> Traceback (most recent call last):
>   File "launch_no_tk.py", line 4, in <module>
>     import pymol
> ModuleNotFoundError: No module named 'pymol'

**Comment by bionicles** [comment]
export PYTHONPATH="/Users/bionhoward/miniconda3/envs/neuromax-2019/bin/"

python
import pymol

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'pymol'
```

**Comment by bionicles** [comment]
`~/hax/<project> $ pymol -qc neuromax.py`

> Traceback (most recent call last):
>   File "/Users/bionhoward/miniconda3/envs/neuromax-2019/lib/python2.7/site-packages/pymol/parsing.py", line 484, in run
>     run_(path, ns_pymol, ns_pymol)
>   File "/Users/bionhoward/miniconda3/envs/neuromax-2019/lib/python2.7/site-packages/pymol/parsing.py", line 533, in run_file
>     execfile(file,global_ns,local_ns)
>   File "/Users/bionhoward/miniconda3/envs/neuromax-2019/lib/python2.7/site-packages/pymol/parsing.py", line 528, in execfile
>     exec(co, global_ns, local_ns)
>   File "neuromax.py", line 3, in <module>

`pip install gym`
> Requirement already satisfied: gym in /usr/local/lib/python3.7/site-packages (0.10.9)


great! so i can get pymol (if I run with it) but then it cannot find the other python packages. if i try to run with python, pymol is nowhere to be found, despite the fact i've added 4 different potential paths to the pythonpath. this sucks

>     from mol import PyMolEnv
>   File "mol.py", line 1, in <module>
>     from gym import spaces

**Comment by bionicles** [comment]
`locate pymol`
just produces a wall of text. no idea which path to use

**Comment by bionicles** [comment]
YAY! 

ok, i found the source of the problem. you MUST specify python=3 when you make your conda env

`$ conda create -n py3 python=3`

the default conda create uses python2.7 i guess, so it installs the wrong version, in the wrong place. 

TLDR: use python=3 conda environment if you want it to find pymol in python3.7 scripts (duh)

**Comment by lorton** [comment]
You beat me to a solution before I could respond.  In general, questions will get more responses using the mailing list:
https://sourceforge.net/p/pymol/mailman/pymol-users/
Or if you're a PyMOL incentive user, help@schrodinger.com will provide professional support.

**Comment by bionicles** [comment]
@lorton thank you,  i will clean this thread so others can more easily find the solution

**Comment by bionicles** [comment]
ugh. after all that. it still doesn't work, because conda installs the paid version, not the open source version

**Comment by cplateVT** [comment]
I received this error when trying to open pymol from the terminal by typing "pymol". The issue was that by default my terminal was opening to an anaconda environment, showing (base) before my current directory. Deactivating the environment by typing: 
"conda deactivate" 
solved the issue. hope this helps somebody !
